persistence/sql/persister_identity.go

```diff
@@ -472,3 +512,110 @@ func (p *Persister) injectTraitsSchemaURL(ctx context.Context, i *identity.Ident
 	i.SchemaURL = s.SchemaURL(p.r.Config(ctx).SelfPublicURL(nil)).String()
 	return nil
 }
+
+func (p *Persister) getJsonSearchQuery(field string, values []string) pop.ScopeFunc {
+	return func(q *pop.Query) *pop.Query {
+		switch p.Connection(context.Background()).Dialect.Name() {
+		case "sqlite3", "mysql", "mariadb":
+			field, innerField := extractFieldAndInnerFields(field)
+			for _, value := range values {
+				if innerField == "" {
+					q = q.Where(fmt.Sprintf(`json_extract(%s, '$') = ?`, field), value)
+				} else {
+					stmt := fmt.Sprintf(`json_extract(%s, '$.%s') = ?`, field, innerField)
+					q = q.Where(stmt, value)
+				}
+			}
+			return q
+		case "postgres", "cockroach":
+			field, innerField := extractFieldAndInnerFields(field)
+			for _, value := range values {
+				if innerField == "" {
+					q = q.Where(fmt.Sprintf(`%s @> '"%s"'`, field, value))
+				} else {
+					q = q.Where(fmt.Sprintf(`%s @> '{"%s":"%s"}'`, field, innerField, value))
+				}
+			}
+			return q
+		}
+		return q
+	}
+}
+
+func (p *Persister) searchCredentialQuery(field string, values []string) pop.ScopeFunc {
+	var innerField string
+	_, innerField = extractFieldAndInnerFields(field)
+	switch innerField {
+	case "type":
+		return func(q *pop.Query) *pop.Query { return q.Where("credential_types.name IN (?)", values) }
+	case "identifier":
+		return func(q *pop.Query) *pop.Query { return q.Where("credential_identifiers.identifier IN (?)", values) }
+	default:
+		return func(q *pop.Query) *pop.Query { return q }
+	}
+}
+
+var quoteChar = map[string]uint8{
+	"cockroach": '"',
+	"mariadb":   '`',
+	"mysql":     '`',
+	"postgres":  '"',
+	"sqlite3":   '"',
+}
+
+func (p *Persister) Quote(ctx context.Context, key string) string {
+	n := p.Connection(ctx).Dialect.Name()
```

Comment:
This is not good enough - there are so many ways to achieve SQL injection. It would also skip any single quote. Please use a strict AlphaNumeric + escaped dot regex here!

Reply:
ok
I'll do it

Reply:
nice :)

Reply:
done